### PR TITLE
feat: Allow to configure stacktrace for captureMessage calls

### PR DESCRIPTION
**Pull request by kamilogorek**
Ref https://github.com/getsentry/raven-node/issues/332

Stack trace example:
screen shot 2017-10-05 at 16 36 03

Removed frame (https://github.com/getsentry/raven-node/compare/message-stacktrace?expand=1#diff-50cfa59973c04321b5da0c6da0fdf4feR337):
screen shot 2017-10-05 at 16 35 59

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -318,8 +319,31 @@ extend(Raven.prototype, {
...
     }
+
     var eventId = this.generateEventId();
-    this.process(eventId, parsers.parseText(message, kwargs), cb);
+
+    if (this.stacktrace) {
+      var ex;
+      // Generate a "synthetic" stack trace
+      try {
+        throw new Error(message);
```
Comment (mitsuhiko) [inline lib/client.js]: do you have to throw the error to get the stack?

Comment (kamilogorek) [reply lib/client.js]: No, it's just more explicit, as we wrap it in `try` clause. Also this is how we write it in `raven-js` and I'm trying to write as unified code as possible.

Comment (mitsuhiko) [reply lib/client.js]: I'm mostly asking because throwing an exception vs just creating an error should have noticable performance differences.

Comment (kamilogorek) [reply lib/client.js]: I'm surprised with this, but you're right : o

```
throw new Error(i)
608.497162ms
575.068622ms
574.888692ms
540.633688ms
540.919783ms
550.634267ms
577.011109ms
541.462129ms
568.272018ms
573.879691ms

avg: 565.1267161000001ms

new Error(i)
410.160736ms
412.978303ms
423.740643ms
434.467241ms
420.66899ms
426.131056ms
430.911071ms
426.683749ms
429.369119ms
432.251688ms

avg: 424.73625960000004ms
```

```js
var start = process.hrtime()
var avg = 0 
var n = 10
var sample = 100000

console.log('\nthrow new Error(i)')

for (var j = 0; j < n; j++) {
  for (var i = 0; i < sample; i++) {
    try {
      throw new Error(i)
    } catch (e) {}
  }

  end = process.hrtime(start)[1]
  start = process.hrtime()

  avg += end 

  console.log('' + (end/1000000) + 'ms')
}
console.log('\navg: ' + (avg/n/1000000) + 'ms')

console.log('\nnew Error(i)')

avg = 0 

for (var j = 0; j < n; j++) {
  for (var i = 0; i < sample; i++) {
    try {
      new Error(i)
    } catch (e) {}
  }

  end = process.hrtime(start)[1]
  start = process.hrtime()

  avg += end 

  console.log('' + (end/1000000) + 'ms')
}
console.log('\navg: ' + (avg/n/1000000) + 'ms')
```js

Comment (kamilogorek) [reply lib/client.js]: Updated

Comment (kamilogorek) [reply lib/client.js]: ~~tests are failing without `throw`~~, in the end it *is* required ¯\_(ツ)_/¯

**Review by mitsuhiko — APPROVED** [review]

**Pull request merged**